src/Compat/FilterProcessor.php

```diff
@@ -112,11 +113,16 @@ protected function requireAndResolveFilterColumns(Filter\Rule $filter, Query $qu
                 $subjectBehaviors = $resolver->getBehaviors($subject);
 
                 // Prepare filter as if it were final to allow full control for rewrite filter behaviors
-                $filter->setValue($subjectBehaviors->persistProperty($filter->getValue(), $columnName));
                 $filter->setColumn($resolver->getAlias($subject) . '.' . $columnName);
                 $filter->metaData()->set('columnName', $columnName);
                 $filter->metaData()->set('relationPath', $path);
 
+                try {
+                    $filter->setValue($subjectBehaviors->persistProperty($filter->getValue(), $columnName));
```

Comment:
Either the comment should also be moved here or the try-catch block should stay in the same place as the line it replaces. I would prefer the latter.